### ISSUE TITLE
The specified delimiter is too long, delimiter should be exactly one character

### ISSUE THREAD

**Issue by alvassin** [iterate-ch/cyberduck#13558]
While uploading files using cyberduck 8.4.1 I get error `Status code: 400, The specified delimiter is too long, delimiter should be exactly one character`. 

On second macbook I had cyberduck 8.2.3 and it worked fine, files were uploaded without any problem, but after update to 8.4.1 i get such error on the second notebook too.

How can i fix that? Attached the screenshot.

![bug](https://user-images.githubusercontent.com/1497444/178162258-73dc0f6b-1c20-417b-96d1-f87c30c824db.jpg)

**Comment by alvassin** [comment]
It seems that instead of correct delimiter filename is passed to delimiter parameter (i was uploading cyberduck.log file):
```
12:24:11.438995+0300 ... GET /?encoding-type=url&max-keys=1&prefix=cyberduck.log%2F&delimiter=cyberduck.log%2F HTTP/1.1
12:24:11.439065+0300 ... Date: Mon, 11 Jul 2022 09:24:11 GMT
12:24:11.439109+0300 ... x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
12:24:11.439152+0300 ... Host: ***.storage.yandexcloud.net
12:24:11.439316+0300 ... x-amz-date: 20220711T092411Z
12:24:11.439404+0300 ... Authorization: ********
12:24:11.439455+0300 ... Connection: Keep-Alive
12:24:11.439496+0300 ... User-Agent: Cyberduck/8.4.1.38006 (Mac OS X/12.2.1) (aarch64)
12:24:11.439536+0300 ... Accept-Encoding: gzip,deflate
12:24:11.479224+0300 ... HTTP/1.1 400 Bad Request
12:24:11.479329+0300 ... Server: nginx
12:24:11.479388+0300 ... Date: Mon, 11 Jul 2022 09:24:11 GMT
12:24:11.479436+0300 ... Content-Type: application/xml; charset=UTF-8
12:24:11.479478+0300 ... Content-Length: 334
12:24:11.479519+0300 ... Connection: close
12:24:11.479559+0300 ... X-Amz-Request-Id: 8e15503a55148d9e
```

**Comment by dkocher** [comment]
This request is used to lookup any existing file and looks legit. Please report this interoperability issue to Yandex as well.

**Comment by alvassin** [comment]
[According to the spec](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html#API_ListObjects_RequestSyntax), delimiter is **a character**, it's valid be 400 response from Yandex and bug is in the Cyberduck client (it is sending more then one character).

**Comment by ylangisc** [comment]
@alvassin Can you please share the full `cyberduck.log` from a failing upload attempt. Unable to replicate it on my end.

**Comment by dkocher** [comment]
The fix of the implementation in #13563 should only apply to directories and thus it is yet unclear why this code path is used in this case.

**Comment by alvassin** [comment]
Please find cyberduck.log attached
[cyberduck.log](https://github.com/iterate-ch/cyberduck/files/9092948/cyberduck.log)
